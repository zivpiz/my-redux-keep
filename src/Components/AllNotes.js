import React, {Component} from 'react';
import {connect} from 'react-redux';
import {SortableContainer, SortableElement} from 'react-sortable-hoc';
import NoteSubmit from './NoteSubmit';
import {dragCheckedNote, dragUncheckedNote} from "../actions";
import './../style.css';


const SortableItem = SortableElement(({note}) => <li>{note}</li>);

const SortableList = SortableContainer(({notes}) => {
    return (
        <ol>
            {notes.map((note, index) => (
                <SortableItem key={index} index={index} note={note}/>
            ))}
        </ol>
    )
});

class AllNotes extends Component {
    constructor(props) {
        super(props);
        this.onUncheckedEnd = this.onUncheckedEnd.bind(this);
        this.onCheckedEnd = this.onCheckedEnd.bind(this);
    }

    onUncheckedEnd = ({oldIndex, newIndex}) => this.props.dragUncheckedNote(oldIndex, newIndex);
    onCheckedEnd = ({oldIndex, newIndex}) => this.props.dragCheckedNote(oldIndex, newIndex);

    render() {
        return (
            <div className="AllNotes">
                <SortableList className="uncheckedNotes" notes={this.props.uncheckedNotes}
                              onSortEnd={this.onUncheckedEnd}/>
                <NoteSubmit/>
                <SortableList className="checkedNotes" notes={this.props.checkedNotes} onSortEnd={this.onCheckedEnd}/>
            </div>
        )
    }
}

const mapStateToProps = state => (
    {
        uncheckedNotes: state.uncheckedNotes,
        checkedNotes: state.checkedNotes
    }
);

export default connect(mapStateToProps, {dragUncheckedNote, dragCheckedNote})(AllNotes);
