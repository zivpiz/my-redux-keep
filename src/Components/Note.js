import React from 'react';
import {connect} from 'react-redux';
import {changeCheckedNote, changeUncheckedNote} from "../actions";
import './../style.css';


const handleChange = (props) => {
    if (props.checked)
        props.changeCheckedNote(props);
    else props.changeUncheckedNote(props);
};

const Note = (props) => {
    return (
        <div className="Note">
            <input type="checkbox" className="NoteCheckbox" onChange={() => handleChange(props)}
                   checked={props.checked}/>
            <label className={`NoteTextChecked-${props.checked}`}>{props.text}</label>
        </div>
    )
};

export default connect(null, {changeCheckedNote, changeUncheckedNote})(Note);