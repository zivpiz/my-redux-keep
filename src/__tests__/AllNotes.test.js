import AllNotes from "../Components/AllNotes";
import Note from "../Components/Note";
import React from 'react';

describe("Test NoteSubmit Component ", ()=>{

    let wrapper;
    const checkedNoteObj = {text:"checked note", checked: true};
    const uncheckedNoteObj = {text:"unchecked note", checked: false};
    const checkedNoteComp = <Note text="Checked Note" checked={true} id={0}/>;
    const uncheckedNoteComp = <Note text="Unchecked Note" checked={false} id={1}/>;

    describe("Snapshot rendering", ()=>{
        beforeEach(()=>{
            wrapper = shallow(<AllNotes />);
        });

        test('Renders appropriate snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("Creation of new notes", ()=>{
        beforeEach(()=>{
            wrapper = shallow(<AllNotes />);
        });

        test('Creates new checked note, appends it to checked notes list and updates id', () => {
            expect(wrapper.find('.uncheckedNotes').children().length).toBe(0);
            expect(wrapper.find('.checkedNotes').children().length).toBe(0);
            const oldId = wrapper.state().nextId;

            wrapper.instance().createNewNote(checkedNoteObj);

            expect(wrapper.find('.uncheckedNotes').children().length).toBe(0);
            expect(wrapper.find('.checkedNotes').children().length).toBe(1);
            expect(wrapper.state().nextId).toBe(oldId + 1);
        });

        test('Creates new unchecked note and appends it to unchecked notes list', () => {
            expect(wrapper.find('.uncheckedNotes').children().length).toBe(0);
            expect(wrapper.find('.checkedNotes').children().length).toBe(0);

            wrapper.instance().createNewNote(uncheckedNoteObj);

            expect(wrapper.find('.uncheckedNotes').children().length).toBe(1);
            expect(wrapper.find('.checkedNotes').children().length).toBe(0);
        });
    });

    describe("Update Check for checked note", ()=>{
        beforeEach(()=>{
            wrapper = shallow(<AllNotes />);
            wrapper.instance().createNewNote(checkedNoteObj);
        });

        test('updates the two lists', () => {
            expect(wrapper.find('.uncheckedNotes').children().length).toBe(0);
            expect(wrapper.find('.checkedNotes').children().length).toBe(1);

            wrapper.instance().updateNoteCheck(checkedNoteComp);

            expect(wrapper.find('.uncheckedNotes').children().length).toBe(1);
            expect(wrapper.find('.checkedNotes').children().length).toBe(0);
        });
    });

    describe("Update Check for unchecked note", ()=>{
        beforeEach(()=>{
            wrapper = shallow(<AllNotes />);
            wrapper.instance().createNewNote(checkedNoteObj);
            wrapper.instance().createNewNote(uncheckedNoteObj);
        });

        test('updates the two lists', () => {
            expect(wrapper.find('.uncheckedNotes').children().length).toBe(1);
            expect(wrapper.find('.checkedNotes').children().length).toBe(1);

            wrapper.instance().updateNoteCheck(uncheckedNoteComp);

            expect(wrapper.find('.uncheckedNotes').children().length).toBe(0);
            expect(wrapper.find('.checkedNotes').children().length).toBe(2);
        });
    });
});


