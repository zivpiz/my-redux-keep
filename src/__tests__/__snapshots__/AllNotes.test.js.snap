// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test NoteSubmit Component  Snapshot rendering Renders appropriate snapshot 1`] = `
<div
  className="AllNotes"
>
  <ol
    className="uncheckedNotes"
  />
  <NoteSubmit
    updateNoteList={[Function]}
  />
  <ol
    className="checkedNotes"
  />
</div>
`;
